### PR TITLE
Upgrade Datepicker to prep for material-ui upgrade

### PR DESCRIPTION
**Pull request by nabramow**
This PR is prep for upgrading Material-UI to v5 (from v4). We need to upgrade our material datepicker package to the newer one and make the breaking changes to the Datepicker.

Breaking this upgrade into smaller PRs so we can better review and make sure we're not introducing any bugs.

TO TEST:

We are using Datepicker several places, make sure they still work ok: Event page selecting event date, sign up for new account, new host request form

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app (No new ones)
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by bakeiro** [comment]
Small detail
icons and text might be to close [host request page]
image

**Comment by bakeiro** [comment]
apart of that minor CSS change the page works good, code looks good

File: app/web/package.json [hunk 1]
```diff
@@ -23,12 +23,15 @@
...
     "@date-io/dayjs": "1.x",
+    "@emotion/react": "^11.13.3",
+    "@emotion/styled": "^11.13.0",
     "@fontsource/mansalva": "^5.1.0",
     "@fontsource/ubuntu": "^5.1.0",
     "@material-ui/core": "^4.12.4",
     "@material-ui/icons": "^4.11.3",
     "@material-ui/lab": "^4.0.0-alpha.61",
-    "@material-ui/pickers": "3.3.10",
+    "@mui/material": "^5.0.0",
```
Comment (bakeiro) [inline app/web/package.json:33]: 🤩

**Review by bakeiro — DISMISSED** [review]

**Comment by bakeiro** [comment]
please before merging do the small CSS change :)

**Review by darrenvong — COMMENTED** [review]
Code looks good to me too 👌 hope I at least helped you to not explore a dead end with my initial approach 🙈

**Comment by nabramow** [comment]
> Code looks good to me too 👌 hope I at least helped you to not explore a dead end with my initial approach 🙈

Yes definitely! If only because once you also couldn't get it to type I finally considered maybe there was something else going on besides me doing it wrong, haha. Teamwork! 🙌

**Comment by nabramow** [comment]
> please before merging do the small CSS change :)

Done!

Desktop
Screenshot 2024-11-09 at 11 03 34 PM

Mobile
Screenshot 2024-11-09 at 11 03 22 PM

**Comment by bakeiro** [comment]
Do you think it's possible to have back the calendar icon? I think looks better, buuuut not a requirement, just a nice to have

image

**Review by bakeiro — APPROVED** [review]

**Comment by nabramow** [comment]
> Do you think it's possible to have back the calendar icon? I think looks better, buuuut not a requirement, just a nice to have
> 

That's not something I changed! It's just how their mobile version is, see here:

https://mui.com/x/react-date-pickers/date-picker/#available-components

I don't think it's worth doing something hacky to change that IMO.

**Pull request merged**